[issue #54] refactored calculateTransform

nickshillingford opened the pull request.

enginmanap commented:
Hi!

Thanks for the PR. I don't think it will compile tough, because the interface is implemented by some classes, and they need to be updated. Then the usage of those should be updated too. I believe compiler will guide you with errors.

nickshillingford commented:
Okay, sounds good. I wasn't sure if you wanted the implementations to be updated as well for this particular issue. I'll update them and submit another pull request.

nickshillingford commented:
Hey,

I updated all the classes I could find that implement the interface, as well as all the use cases of the method I could find. Compiled for me.

Also, I changed the bool name from "isFound" to "status" since you said that it should return the status.

enginmanap commented on src/Assets/Animations/AnimationCustom.cpp:
```diff
@@ -8,16 +8,18 @@
-    resultTransformation.setTranslate( animationNode->getPositionVector(time));
-    resultTransformation.setOrientation(animationNode->getRotationQuat(time));
+bool AnimationCustom::calculateTransform(const std::string& nodeName __attribute((unused)), float time, Transformation& transformation) const {
+    bool status = false;
```
Since status is not a parameter, you can just return it, without creating a variable

enginmanap commented on src/World.cpp:
```diff
@@ -207,8 +207,8 @@ World::World(const std::string &name, PlayerInfo startingPlayerType, InputHandle
-                bool isFound;
-                Transformation tf = animationCustom->calculateTransform("", animationTime, isFound);
+                Transformation tf;
+                bool status = animationCustom->calculateTransform("", animationTime, tf);
```
if not used, status can be removed.

enginmanap commented on src/Assets/Animations/AnimationCustom.cpp:
```diff
@@ -8,16 +8,18 @@
-    resultTransformation.setOrientation(animationNode->getRotationQuat(time));
+bool AnimationCustom::calculateTransform(const std::string& nodeName __attribute((unused)), float time, Transformation& transformation) const {
+    bool status = false;
+    if (nodes.find(nodeName) == nodes.end()) {
```
There is no nodes map, and it is not used in custom animations. that part can be removed completely.

nickshillingford commented:
Hmm, I don't know why AnimationSequencer didn't appear when I did a scan of the repository for calculateTransform uses. I'm sorry about missing that. 

I am not at home currently but I will definitely fix that and remove the unnecessary code by the end of the day. Is that okay?

enginmanap commented:
No need to rush, you can do whenever you like :)

nickshillingford commented:
Fixed AnimationSequencer.cpp, removed the unnecessary bool variable from World.cpp and the map search from AnimationCustom.cpp, since there is no map instance for that class.

(I meant to say unnecessary, not necessary, in the description for commit 58e7b09)

Everything should work now. Is it good?

enginmanap commented:
Hey,

I just merged it, great work!

nickshillingford commented:
Awesome!
